build/gulp/tasks/perf.ts

```diff
@@ -0,0 +1,117 @@
+import * as express from 'express'
+import * as fs from 'fs'
+import { task, series } from 'gulp'
+import * as _ from 'lodash'
+import * as puppeteer from 'puppeteer'
+import * as rimraf from 'rimraf'
+import { argv } from 'yargs'
+
+import { ProfilerMeasure, ProfilerMeasureCycle } from '../../../perf/types'
+import config from '../../../config'
+import webpackPlugin from '../plugins/gulp-webpack'
+
+const { paths } = config
+const { colors, log } = require('gulp-load-plugins')().util
+
+let server
+
+const floor = (value: number) => _.floor(value, 2)
+
+const computeMeasureMedian = (measures: ProfilerMeasure[], key: string) => {
+  const values = _.sortBy(measures, key)
+
+  const lowMiddle = Math.floor((values.length - 1) / 2)
+  const highMiddle = Math.ceil((values.length - 1) / 2)
+
+  return (values[lowMiddle][key] + values[highMiddle][key]) / 2
+}
+
+const reduceMeasures = (measures: ProfilerMeasure[], key: string) => {
+  let min = measures[0][key]
+  let max = measures[0][key]
+  let sum = measures[0][key]
+
+  for (let i = 1; i < measures.length; i++) {
+    if (measures[i][key] < min) min = measures[i][key]
+    if (measures[i][key] > max) max = measures[i][key]
+
+    sum = sum + measures[i][key]
+  }
+
+  return {
+    avg: floor(sum / measures.length),
+    median: floor(computeMeasureMedian(measures, key)),
+    min: floor(min),
+    max: floor(max),
+    values: _.map(measures, measure => ({
+      order: measure.order,
+      value: measure[key],
+    })),
+  }
+}
+
+const normalizeMeasures = (measures: ProfilerMeasureCycle[]) => {
+  const perExampleMeasures = _.reduce(
+    measures,
+    (result, cycle: ProfilerMeasureCycle) => {
+      _.forEach(cycle, (value: ProfilerMeasure, exampleName: string) => {
+        result[exampleName] = [...(result[exampleName] || []), value]
+      })
+
+      return result
+    },
+    {},
+  )
+
+  return _.mapValues(perExampleMeasures, (measures: ProfilerMeasure[]) => ({
+    actualTime: reduceMeasures(measures, 'actualTime'),
+    baseTime: reduceMeasures(measures, 'baseTime'),
+  }))
+}
+
+task('perf:clean', cb => {
+  rimraf(paths.perfDist(), cb)
+})
+
+task('perf:build', cb => {
+  webpackPlugin(require('../../../build/webpack.config.perf').default, cb)
+})
+
+task('perf:run', async () => {
+  const measures: ProfilerMeasureCycle[] = []
+  const times = argv.times || 10
+
+  const browser = await puppeteer.launch()
+
+  for (let i = 0; i < times; i++) {
+    const page = await browser.newPage()
+    await page.goto(`http://${config.server_host}:${config.server_port}`)
+
+    const measuresFromStep = await page.evaluate(() => window.runMeasures())
+    await page.close()
```

Comment:
would suggest to put it into `finally` block and, thus, guarantee that `page` process resources will be reclaimed no matter what

Reply:
We don't handling any errors there at all, not sure that we should do this now 💭

Reply:
would still suggest that, as those are a bit different types of problems
- if those that we are not handling are about some correctness problems, they won't have global effect to the platform process is running on
- here, in contrast, this global effect might take place in case if OS resources are consumed without being reclaimed on failure

Not saying that we should introduce error handling measures everywhere, but here see this as a very important case we need to address now

Reply:
Added error handling, please check 👍